state/execution_test.go

```diff
@@ -233,7 +233,7 @@ func TestProcessProposal(t *testing.T) {
 
```

Comment:
The original PR introduced some improvements to test case `TestProcessProposal`.
Consider bringing these improvements along (a few 10s of lines)